daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -1155,3 +1156,23 @@ class DaffodilUnparseContentHandler private[japi] (sContentHandler: SDaffodilUnp
   override def skippedEntity(name: String): Unit =
     contentHandler.skippedEntity(name)
 }
+
+/**
+ * Returns the EntityResolver used by Daffodil to resolve import/include
+ * schemaLocations.
+ *
+ * DFDLCatalogResolver uses an org.apache.xml.resolver.Catalog/CatalogManager to
+ * attempt to resolve namespaces and systemId's. If we are unable to resolve a
+ * file with the Catalog, we attempt to resolve the file using the systemId path
+ * from the schemaLocation attribute. If this fails we throw a
```

Comment:
> we attempt to resolve the file using the systemId path from the schemaLocation attribute.

Does this require a little extra clarification? Don't we look at both the classpath and the filesystem, and if so, which do we do first? Is it worth mentioning relative paths in schemaLocations are relative to the file that includes (it I think that's right)?

This might also work better as a numeric list, e.g.

```
The returned resolver looks in the following places to resolve entities in this order:

1. org.apache....CataogManager  ...
2. Java classpath ...
3. Filesystem ...
```

Reply:
Actually, looking through our code I don't think we technically reach out to the filesystem at all.  We only search for things on the classpath or relative to some context URI t hat gets passed in

Reply:
Ah, I guess that makes sense, and it just depends on if the root schema was found in on the classpath or the filesystem. If the root schema was found on the classpath all imports will look on the classpath (since they are relative to things found on the classpath), and likewise with the filesystem.

So maybe instead of mentioning the classpath we say something like, catalogmanager is first, otherwise schemaLocations are resolves relative to the importing schema URI, which could be a file on the filesystem or in a jar on the classpath. Probably can be worded better, but something like that to make it clear we support classpaths or file depending on the original root schema.